### PR TITLE
Integrate GitHub API

### PR DESCRIPTION
**Pull request by yashre-bh**

### CODE REVIEW

File: lib/api/application.go [hunk 1]
```diff
@@ -134,6 +95,21 @@ func SetupApplication(app types.Application) types.ResponseError {
 		}
 	}
 
+	_, err = docker.ExecProcess(app.GetContainerID(), []string{"git", "init"})
+	if err != nil {
+		return types.NewResErr(500, "git init unsuccessful", err)
```
Comment (mhk19) [inline lib/api/application.go:100]:
```suggestion
		setup <- types.NewResErr(500, "git init unsuccessful", err)
		return
```
This should be like this

File: lib/api/application.go [hunk 2]
```diff
@@ -134,6 +95,21 @@ func SetupApplication(app types.Application) types.ResponseError {
...
+	}
+
+	_, err = docker.ExecProcess(app.GetContainerID(), []string{"git", "remote", "add", "origin", app.GetGitRepositoryURL()})
+	if err != nil {
+		return types.NewResErr(500, "setting remote unsuccessful", err)
```
Comment (mhk19) [inline lib/api/application.go:105]: same as above

File: lib/api/application.go [hunk 3]
```diff
@@ -134,6 +95,21 @@ func SetupApplication(app types.Application) types.ResponseError {
...
+	}
+
+	_, err = docker.ExecProcess(app.GetContainerID(), []string{"git", "pull", "origin", app.GetGitRepositoryBranch()})
+	if err != nil {
+		return types.NewResErr(500, "pulling contents unsuccessful", err)
```
Comment (mhk19) [inline lib/api/application.go:110]: same as above

File: services/appmaker/controller.go [hunk 4]
```diff
@@ -41,15 +41,15 @@ func (s *server) Create(ctx context.Context, body *pb.RequestBody) (*pb.Response
...
 	rateCount := configs.ServiceConfig.RateLimit
 	timeInterval := configs.ServiceConfig.RateInterval
-	if user.IsAdmin() == false && maxCount >= 0 {
+	if !user.IsAdmin() && maxCount >= 0 {
 		rateLimitCount := mongo.CountInstanceInTimeFrame(body.GetOwner(), mongo.AppInstance, timeInterval)
 		totalCount := mongo.CountInstancesByUser(body.GetOwner(), mongo.AppInstance)
 		if totalCount < maxCount {
 			if rateLimitCount >= rateCount && rateCount >= 0 {
-				return nil, fmt.Errorf("Cannot deploy more than %d app instances in %d hours", rateCount, timeInterval)
+				return nil, fmt.Errorf("cannot deploy more than %d app instances in %d hours", rateCount, timeInterval)
```
Comment (mhk19) [inline services/appmaker/controller.go:49]: why this change

File: services/appmaker/controller.go [hunk 5]
```diff
@@ -41,15 +41,15 @@ func (s *server) Create(ctx context.Context, body *pb.RequestBody) (*pb.Response
...
 			}
 		} else {
-			return nil, fmt.Errorf("Cannot deploy more than %d app instances", maxCount)
+			return nil, fmt.Errorf("cannot deploy more than %d app instances", maxCount)
```
Comment (mhk19) [inline services/appmaker/controller.go:52]: why this change

File: services/appmaker/controller.go [hunk 6]
```diff
@@ -70,7 +70,7 @@ func (s *server) Create(ctx context.Context, body *pb.RequestBody) (*pb.Response
 	}
 
 	if pipeline[language] == nil {
-		return nil, fmt.Errorf("Language `%s` is not supported", language)
+		return nil, fmt.Errorf("language `%s` is not supported", language)
```
Comment (mhk19) [inline services/appmaker/controller.go:73]: why this change

File: services/appmaker/controller.go [hunk 7]
```diff
@@ -142,33 +142,18 @@ func (s *server) Create(ctx context.Context, body *pb.RequestBody) (*pb.Response
...
 		return nil, err
 	}
 
-	diskCleanup(appName)
-
-	if pipeline[app.Language] == nil {
-		return nil, fmt.Errorf("Non-supported language `%s` specified for `%s`", app.Language, appName)
-	}
+	pullChanges := []string{"git", "pull", "origin", app.GetGitRepositoryBranch()}
+	_, err = docker.ExecProcess(app.ContainerID, pullChanges)
```
Comment (mhk19) [inline services/appmaker/controller.go:151]: are we sure that the container will be up?

**Review by mhk19 — CHANGES_REQUESTED** [review]

File: services/appmaker/controller.go (hunk 4 above)
Comment (yashre-bh) [reply services/appmaker/controller.go:49]: error strings should not be capitalized (ST1005) go-staticcheck: the linter for go suggested this change

File: services/appmaker/controller.go (hunk 5 above)
Comment (yashre-bh) [reply services/appmaker/controller.go:52]: error strings should not be capitalized (ST1005) go-staticcheck: the linter for go suggested this change

File: services/appmaker/controller.go (hunk 6 above)
Comment (yashre-bh) [reply services/appmaker/controller.go:73]: error strings should not be capitalized (ST1005) go-staticcheck: the linter for go suggested this change

File: lib/api/application.go (hunk 1 above)
Comment (yashre-bh) [reply lib/api/application.go:100]: [here](https://github.com/sdslabs/gasper/blob/44c8d3adcd6bb84698a20e3d3fbe388a54402669/lib/api/application.go#L121)

This is the function definition (as on the develop branch).
`setup channel` was not passed as a parameter in `SetupApplication()` function, I did not alter this. Will I still have to make this change?

**Review by mhk19 — APPROVED** [review]

**Pull request merged**